pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -602,6 +629,22 @@ class CreateVmModal extends React.Component {
                 this.setState({ storagePool: "NewVolume" });
             }
             break;
+        case 'os': {
+            this.setState(prevState => { // to prevent asynchronous calls
```

Comment:
You are using the functional setState without actually utilizing it's format.
Aka you have prevState available but you are accessing state via this.state.

Reply:
fixed